framework/base/Model.php

```diff
@@ -356,6 +356,10 @@ public function validate($attributeNames = null, $clearErrors = true)
             $attributeNames = $this->activeAttributes();
         }
 
+        if (!is_array($attributeNames)) {
+            $attributeNames = (array)$attributeNames;
+        }
+
```

Comment:
You could remove `if` and always cast to array.

Reply:
Done, thank you, it helped to solve codeclimate issue